packages/@aws-cdk/aws-cloudfront/lib/distribution.ts

```diff
@@ -1,14 +1,302 @@
+    }
+
+    this.certificate = props.certificate;
+    this.errorConfigurations = props.errorConfigurations ?? [];
+
+    const distribution = new CfnDistribution(this, 'CFDistribution', { distributionConfig: {
+      enabled: true,
+      origins: Lazy.anyValue({ produce: () => this.renderOrigins() }),
+      defaultCacheBehavior: this.defaultBehavior._renderBehavior(),
+      cacheBehaviors: Lazy.anyValue({ produce: () => this.renderCacheBehaviors() }),
+      viewerCertificate: this.certificate ? { acmCertificateArn: this.certificate.certificateArn } : undefined,
+      customErrorResponses: this.renderCustomErrorResponses(),
+      priceClass: props.priceClass ?? undefined,
+    } });
+
+    this.domainName = distribution.attrDomainName;
+    this.distributionId = distribution.ref;
+  }
+
+  /**
+   * Adds a new behavior to this distribution for the given pathPattern.
+   *
+   * @param pathPattern the path pattern (e.g., 'images/*') that specifies which requests to apply the behavior to.
+   * @param behaviorOptions the options for the behavior at this path.
+   */
+  public addBehavior(pathPattern: string, behaviorOptions: BehaviorOptions) {
+    if (pathPattern === '*') {
+      throw new Error('Only the default behavior can have a path pattern of \'*\'');
+    }
+    this.additionalBehaviors.push(new CacheBehavior({ pathPattern, ...behaviorOptions }));
+  }
+
+  private renderOrigins(): CfnDistribution.OriginProperty[] {
+    const origins = new Set<Origin>();
+    origins.add(this.defaultBehavior.origin);
+    this.additionalBehaviors.forEach(behavior => origins.add(behavior.origin));
+
+    const renderedOrigins: CfnDistribution.OriginProperty[] = [];
+    origins.forEach(origin => renderedOrigins.push(origin._renderOrigin()));
+    return renderedOrigins;
+  }
+
+  private renderCacheBehaviors(): CfnDistribution.CacheBehaviorProperty[] | undefined {
+    if (this.additionalBehaviors.length === 0) { return undefined; }
+    return this.additionalBehaviors.map(behavior => behavior._renderBehavior());
+  }
+
+  private renderCustomErrorResponses(): CfnDistribution.CustomErrorResponseProperty[] | undefined {
+    if (this.errorConfigurations.length === 0) { return undefined; }
+    function validateCustomErrorResponse(errorResponse: CfnDistribution.CustomErrorResponseProperty) {
+      if (errorResponse.responsePagePath && !errorResponse.responseCode) {
+        throw new Error('\'responseCode\' must be provided if \'responsePagePath\' is defined');
+      }
+      if (!errorResponse.responseCode && !errorResponse.errorCachingMinTtl) {
+        throw new Error('A custom error response without either a \'responseCode\' or \'errorCachingMinTtl\' is not valid.');
+      }
+    }
+    this.errorConfigurations.forEach(e => validateCustomErrorResponse(e));
+    return this.errorConfigurations;
+  }
+
+}
+
+/**
+ * The price class determines how many edge locations CloudFront will use for your distribution.
+ */
+export enum PriceClass {
```

Comment:
Note: These were moved from the `web_distribution` file. Could cause breakage for customers who have imported directly by filename.

Reply:
Guess we can live with that.